src/adapters/mod.rs

```diff
@@ -5,20 +5,38 @@
 mod philips_hue;
 
 use controller::Controller;
+use self::philips_hue::PhilipsHueAdapter;
 use service::ServiceAdapter;
+use std::sync::{ Arc, Mutex };
 
 pub struct AdapterManager<T> {
-    controller: T
+    controller: T,
+    adapters: Arc<Mutex<Vec<Box<ServiceAdapter>>>>
```

Comment:
Why do we need an Arc<Mutex<>> here? AdapterManager is always used on the same thread.